src/main/java/jenkins/plugins/mailer/tasks/MimeMessageBuilder.java

```diff
@@ -212,12 +212,16 @@ private void addBody(MimeMessage msg) throws MessagingException {
             Multipart multipart = new MimeMultipart();
             BodyPart bodyPart = new MimeBodyPart();
 
-            bodyPart.setContent(body, mimeType);
+            bodyPart.setContent(body, contentType());
             multipart.addBodyPart(bodyPart);
             msg.setContent(multipart);
         }
     }
 
+    private String contentType() {
+        return String.format("%s; charset=%s", mimeType, charset);
```

Comment:
After checking how `MimeBodyPart` is implemented, it might make sense to tweak this to:

``` java
    return String.format("%s; charset=%s", mimeType, MimeUtility.quote(charset, HeaderTokenizer.MIME));	
```